runtime/utils/common/src/aws/smithy/kotlin/runtime/util/text/Text.kt

```diff
@@ -99,7 +99,7 @@ private const val upperHex: String = "0123456789ABCDEF"
 private val upperHexSet = upperHex.toSet()
 
 // $2.1 Percent-Encoding
-private fun Byte.percentEncode(): String = buildString(3) {
+fun Byte.percentEncode(): String = buildString(3) {
```

Comment:
This should be marked `@InternalApi` 

We also might consider changing this to two different functions:

```kotlin

@InternalApi
fun Byte.percentEncodeTo(out: Appendable): Unit {
    val code = toInt() and 0xff
    out.append('%')
    out.append(upperHex[code shr 4])
    out.append(upperHex[code and 0x0f])
}

@InternalApi
private fun Byte.percentEncode(): String {
    val byte = this
    return StringBuilder(3)
        .apply { byte.percentEncodeTo(this) }
        .toString()

}
```

This would allow percent encoding to append directly to an existing string builder (e.g. in our existing encode functions)